code/game/objects/items/weapons/storage/storage.dm

```diff
@@ -364,6 +364,12 @@
 /obj/item/weapon/storage/proc/handle_item_insertion(obj/item/W as obj, prevent_warning = 0)
 	if(!istype(W))
 		return 0
+
+	//Allow smashing of storage items on harm intent without also putting the weapon into the container.
+	var/mob/user = usr
+	if (user.a_intent == I_HURT && breakable_flags & BREAKABLE_WEAPON && loc != user)
```

Comment:
Runtimes if `usr` does not exist. Why not just put it under the `if(usr)` right below?